posthog/migrations/0288_add_tags_to_feature_flags.py

```diff
@@ -0,0 +1,100 @@
+# Generated by Django 3.2.16 on 2023-01-10 22:50
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("posthog", "0287_add_session_recording_model"),
+    ]
+
+    operations = [
+        migrations.RemoveConstraint(
+            model_name="taggeditem",
+            name="exactly_one_related_object",
+        ),
+        migrations.AddField(
+            model_name="taggeditem",
+            name="feature_flag",
+            field=models.ForeignKey(
+                blank=True,
```

Comment:
@alexkim205 Getting migration error "field is not null and will lock the table will migrating" even though I'm using blank + null = True, do you have any idea on why this is the case?

Reply:
I don't see anything wrong with the migration and the error message might be misleading. My guess is that the regex for checking if our migrations are safe is getting caught on this line of the migration, thinking that we're adding a new field here that is not null.

```
CREATE UNIQUE INDEX "unique_feature_flag_tagged_item" ON "posthog_taggeditem" ("tag_id", "feature_flag_id") WHERE "feature_flag_id" IS NOT NULL;
```

We already added this field before with 
```
ALTER TABLE "posthog_taggeditem" ADD COLUMN "feature_flag_id" integer NULL CONSTRAINT 
```

I haven't been able to test it locally but would changing this line (https://github.com/PostHog/posthog/blob/45b4f7092050e606454370cdd019b3b15de23c75/posthog/management/commands/test_migrations_are_safe.py#L17-L17) to:

```
re.findall(r"(?<!DROP|CREATE UNIQUE INDEX) (NOT NULL|DEFAULT)", sql, re.M & re.I)
```

make sure we're not catching these false positives in the test?

Reply:
^^ Migration check is too aggressive